node/src/components/storage.rs

```diff
@@ -1571,6 +1571,27 @@ impl Storage {
             .transpose()
     }
 
+    /// Retrieves a single complete block header by height by looking it up in
+    /// the index and returning it.
+    pub fn read_complete_block_header_by_height(
```

Comment:
Would it be possible to use the already existing [get_block_header_by_height_restricted()](https://github.com/casper-network/casper-node/blob/a54d19ba1e4484d26961a1f48da3ba8bffa38322/node/src/components/storage.rs#L1948)?

Reply:
It's certainly possible, but I didn't want to do that mainly because `get_block_header_by_height_restricted` looks like a function which may change in the future (I'm talking about the restrictions) and I wanted something explicit here. I have no strong opinion on it though, we can safely use `get_block_header_by_height_restricted` if it's better.

Reply:
Forgot to mention in my previous comment, `get_block_header_by_height_restricted` also requires a transaction instance (which is probably why it's not a `pub fn`) and I don't want to create those outside of storage.

Reply:
Yes, I agree there are differences. Despite not liking this code duplication much, I'm not strong about it. So let it be :)

Anyway, I also see that the only restriction we now have is `only_from_available_block_range`, which is set to `true` in all the flows. This is probably a remnant of the previous implementation of fast sync and we can as well get rid of it, hitting two birds with one stone. That'd be something to consider as a separate ticket (cc: @Fraser999)

Reply:
Happy for that suggestion to get folded into this PR, or if not @georgepisaltu let me know and we can create a ticket request.

Reply:
We will need to create a separate ticket for this cleanup. I tried to do it but it's not as simple as it looks because "only from available block range" is not the same as "complete". There are multiple usages of these functions in storage and I believe they are due for a cleanup, but it would take more investment than it would make sense for the issue this PR solves.